src/lib/connectors/jsdom/jsdom.ts

```diff
@@ -226,7 +226,7 @@ class JSDOMConnector implements IConnector {
      * * uses `favicon.ico` and the final url after redirects.
      */
     private async getFavicon(element?: HTMLElement) {
-        const href = element ? element.getAttribute('href') : '/favicon.ico';
+        const href = (element && element.getAttribute('href')) || '/favicon.ico';
```

Comment:
@sarvaje This is due to the scenario when `href` is an empty string. We still want to send out a request to `/favicon.ico` in this case.